### PR TITLE
Improve recommended requirements FAQ answer

### PR DESCRIPTION
**Pull request by react-learner**
Based on feedback from @qdequele .

### CODE REVIEW

File: create/faq.md [hunk 1]
```diff
@@ -180,20 +180,43 @@ Until our first stable release (v1.0), MeiliSearch minor versions are not compat
 
 ## What are the recommended requirements for hosting a MeiliSearch instance?
 
-**Short answer**: It's impossible to say. We recommend [experimenting with RAM availability and database size](#how-does-hardware-affect-search-speed) to find the hardware + relevancy setup that works best for your dataset.
+### Short Answer
 
-**Long answer**: The disk size and RAM usage of your MeiliSearch database can vary widely. The following factors have a great effect on space and memory use:
+It's extremely difficult to accurately estimate the size and memory use of a MeiliSearch database, even knowing the size of the `.JSON` files. There's just too much complexity involved in the indexing process. For a benchmark and more thorough explanation, [go here](/reference/under_the_hood/storage.md#measured-disk-usage).
+
+We recommend reading [the long answer below](#long-answer) to find the setup that works best for your needs.
```
Comment (fharper) [inline create/faq.md:187]: Do we need that sentence since right after, there is the "long answer" header.

File: create/faq.md [hunk 2]
```diff
@@ -180,20 +180,43 @@ Until our first stable release (v1.0), MeiliSearch minor versions are not compat
...
+
+### Long Answer
+
+#### Memory Use
+
+When it comes to RAM, there are two things that can cause your usage to spike:
```
Comment (fharper) [inline create/faq.md:193]: Wen it comes to memory usage
or
Wen it comes to memory usage (RAM)

File: create/faq.md [hunk 3]
```diff
@@ -180,20 +180,43 @@ Until our first stable release (v1.0), MeiliSearch minor versions are not compat
...
+
+1. Adding documents
+2. Updating index settings (if index contains documents)
+
+To reduce RAM usage, follow this best practice: always update index settings **before** adding your documents.
```
Comment (fharper) [inline create/faq.md:198]: To reduce memory usage

Some people may not know what is ram, many cloud provider just write "Memory 8Gb"

Comment (fharper) [inline create/faq.md:198]: What happen if you don't? Would be nice to give more explanation on why they should do that.

File: create/faq.md [hunk 4]
```diff
@@ -180,20 +180,43 @@ Until our first stable release (v1.0), MeiliSearch minor versions are not compat
...
+
+:::tip
+MeiliSearch's new engine (release date: sometime in 2021) will allow you to set a limit on how much RAM MeiliSearch uses.
```
Comment (fharper) [inline create/faq.md:201]: Maybe "release date: sometime in 2021" could be "before the end of 2021"? Maybe it's just me, but "sometime in 2021" feels like we are not organized as "before the end..." is more like, we know when, but we don't make it public yet.

File: create/faq.md [hunk 5]
```diff
@@ -180,20 +180,43 @@ Until our first stable release (v1.0), MeiliSearch minor versions are not compat
...
+:::
+
+#### Disk Use
+
+The following factors have a great impact on the size of your database (in no particular order):
```
Comment (fharper) [inline create/faq.md:206]: I think I made the same comment elsewhere, but can we order them or it's impossible for us to order them?

File: create/faq.md [hunk 6]
```diff
@@ -180,20 +180,43 @@ Until our first stable release (v1.0), MeiliSearch minor versions are not compat
...
+- The number of searchable fields
 - The number of faceted fields
 - The size of each update
-- The number of different words present in documents (beware heavily multi-lingual datasets and fields with unique words, such as IDs)
-
-All of this means that **it's almost impossible to estimate the size and memory usage of a MeiliSearch database** before creation, even based on the size of its documents. There are, however, a few [general principles](#how-does-hardware-affect-search-speed) you should know.
+- The number of different words present in the dataset
+  
+:::tip
+Beware heavily multi-lingual datasets and datasets with many unique words, such as IDs or URLs. If you do have ID or URL fields, [make them non-searchable](/reference/api/searchable_attributes.md#update-searchable-attributes).
```
Comment (fharper) [inline create/faq.md:216]: make them non-searchable... if they aren't useful as a search criteria?

IDs I understand, but someone may want to let the user search URL as an example, so I don't want people to think we tell them not to make fields they need non searchable.

Comment (fharper) [inline create/faq.md:216]: Maybe it's just me but I feel like the sentence isn't ended...

Beware heavily multi-lingual datasets and datasets with many unique words, such as IDs or URLs ... as they will slow down the search and make your database size bigger?

Something like that

File: create/faq.md [hunk 7]
```diff
@@ -180,20 +180,43 @@ Until our first stable release (v1.0), MeiliSearch minor versions are not compat
...
+:::
 
-## How does hardware affect search speed?
+#### How does hardware affect search speed?
```
Comment (fharper) [inline create/faq.md:219]: Are you sure? They are a question themselves, previous was is a H2.

Unless you want to move them under the "What are the recommended requirements for hosting a MeiliSearch instance?" question. If so, I would change the title for "Search Speed" and add the "Speeding Up MeiliSearch" under.

**Review by fharper — CHANGES_REQUESTED** [review]

File: create/faq.md (hunk 7 above)
Comment (react-learner) [reply create/faq.md:219]: You're right, I decided that all of this information should be grouped under the same "Question" so it's easy to find, but in that case I should change the heading so it's not posing a separate question.

File: create/faq.md [hunk 8]
```diff
@@ -180,20 +180,41 @@ Until our first stable release (v1.0), MeiliSearch minor versions are not compat
...
+
+There are two things that can cause your memory usage (RAM) to spike:
+
+1. Adding documents
+2. Updating index settings (if index contains documents)
+
+To reduce memory use and indexing time, follow this best practice: **always update index settings before adding your documents**. This avoids unnecessary double-indexing.
+
+:::tip
+MeiliSearch's new engine (release date: before the end of 2021) **will allow you to set a limit on how much RAM MeiliSearch uses**.
```
Comment (curquiza) [inline create/faq.md:199]: The user will not be able to set the RAM usage once the new search engine is released, it will come later 🙂

Comment (react-learner) [reply create/faq.md:199]: Didn't know that, thanks @curquiza ! Will delete this sentence.

**Review by bidoubiwa — APPROVED** [review]
🔥🔥🔥🔥

**Comment by curquiza** [comment]
Hello, this is not a big deal but ff I'm not wrong this sentence was not removed as I required -> "MeiliSearch's new engine (release date: before the end of 2021) **will allow you to set a limit on how much RAM MeiliSearch uses**."

**Comment by react-learner** [comment]
@curquiza See https://github.com/meilisearch/documentation/pull/921#pullrequestreview-648081961 😄

**Comment by curquiza** [comment]
Thanks 😁

**Pull request merged**